frontend/src/functions.js

```diff
@@ -10,251 +11,284 @@ function radians_to_degrees(radians) {
   for (let i = 0; i < replaceEditMatchArray.length; i++) {
     outputState[replaceEditMatchArray[i].clientName] = replaceEditMatch(
       apiState,
-      outputState,
       replaceEditMatchArray[i].serverName,
       replaceEditMatchArray[i].clientName,
       replaceEditMatchArray[i].arrayToMapTrue,
       replaceEditMatchArray[i].serverKeys,
       replaceEditMatchArray[i].clientKeys
     );
   }
+  return outputState;
+}
 
-  console.log(outputState);
+function apiUpdateState(that) {
+  axios.get(constants.serverURL + `/state`).then(data => {
+    // take the returned data and translate it
+    let translatedState = translateState(data.data);
+    // CALLBACK CHECKS
 
-  /*
-  return {
-    players: {
-      PLYR1: {
-        name: "Lightning Bug, LLC.",
-        color: "Red"
-      },
-      PLYR2: {
-        name: "Oil Barron, Inc.",
-        color: "Black"
-      },
-      PLYR3: {
-        name: "Armco, Co.",
-        color: "Blue"
-      }
-    },
-    playerOrder: ["PLYR2", "PLYR3", "PLYR1"],
-    currentPlayer: "PLYR1",
-    ownedPowerPlants: {
-      PLYR1: ["1", "2", "3"],
-      PLYR2: ["4", "5", "6"],
-      PLYR3: ["7", "8", "9"]
-    },
-    ownedResources: {
-      PLYR1: {
-        money: 40,
-        coal: 5,
-        garbage: 2,
-        uranium: 1,
-        oil: 3
-      },
-      PLYR2: {
-        money: 20,
-        coal: 1,
-        garbage: 0,
-        uranium: 3,
-        oil: 2
-      },
-      PLYR3: {
-        money: 95,
-        coal: 5,
-        garbage: 1,
-        uranium: 5,
-        oil: 10
-      }
-    },
-    ownedCities: {
-      PLYR1: ["portland"],
-      PLYR2: ["boise"],
-      PLYR3: ["billings"]
-    },
-    actualMarket: ["18", "19", "20", "21"],
-    futureMarket: ["30", "31", "28", "25"],
-    gamePhase: "auctionBidding", // "auctionPickPlant", "auctionBidding",
-    currentBid: 0,
-    highestBidder: "PLYR3"
-  };
-  */
-}
+    // 1. Check to see if we are entering auctionPickPlant so we can pop the modal up
+    if (
+      translatedState.gamePhase !== "auctionPickPlant" &&
+      translatedState.gamePhase === "auctionPickPlant" &&
+      that.state.showPowerPlantMarket === false
+    ) {
+      that.setState({
+        showPowerPlantMarket: true
+      });
+    }
+
+    // CALLBACK CHECKS COMPLETE
 
-function apiUpdateState() {
-  axios.get(`http://localhost:8080/state`).then(data => {
-    translateState(data.data);
-    // create players object
+    // Update the gameState
+    that.setState({
+      gameState: translatedState
+    });
   });
 }
 
-function apiCreateUser() {}
+function apiUpdateStateInterval(that) {
+  that.setState(
+    {
+      userProfile: constants.emptyClientGameStateSchema.userProfile,
+      gameState: constants.emptyGameStateSchema
+    },
+    () => {
+      // after first state is set, setInterval to ping gameState API every 100ms
+      that.interval = setInterval(function() {
+        apiUpdateState(that);
+        // do every 100ms
+      }, 100);
```

Comment:
If the network call ever takes longer than 100ms to complete, then things could get a little weird.

Not a concern in a toy app like this.